MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs

```diff
@@ -0,0 +1,1144 @@
+                            IWebElement startenddateerrormessage = driver.FindElement(By.XPath(startenddateerrormessageXP));
+                            if (startenddateerrormessage.Text == StartEndDateError)
+                            {
+                                Console.WriteLine("Actual start date / end date error message '" + startenddateerrormessage.Text + "' matches expected error");
+                                test.Log(Status.Pass, "Actual start date / end date error message '" + startenddateerrormessage.Text + "' matches expected error");
+                            }
+                            else
+                            {
+                                Console.WriteLine("Actual start date / end date error message '" + startenddateerrormessage.Text + "' does not match expected error '" + StartEndDateError);
+                                test.Log(Status.Fail, "Actual start date / end date error message '" + startenddateerrormessage.Text + "' does not match expected error '" + StartEndDateError + "'");
+                            }
+                        }
+                    }
+                } // Save button was not found so presumed to be on the Manage Listings page
+            }
+
+            if (ActionButton == "Cancel")
+            {
+                // click the Cancel button
+                cancelButton.Click();
+
+                // user is taken back to the page where they were when they clicked the Share Skill button
+
+                // set savedMessage to empty because no pop-up message appears which would need to be verified
+                //savedMessage = "";
+            }
+
+            //return savedMessage;
+            
+            // if ValidInvalid parameter = Valid, then message should be successful, otherwise; the message will be unsuccessful
+            // must be able to handle test case where invalid data are corrected and successfully saved in same test case
+        }
+
+        public string Reformat3TimeElements(string TimetoReformat)
+        {
+            // return 6 character string containing: 2-digit hour, 2-digit minutes, and 2 character AM/PM
+
+            timeString = "";
+
+            if (TimetoReformat.Substring(1, 1) == ":")
+            {
+                timeString = ("0" + TimetoReformat.Substring(0, 1) + TimetoReformat.Substring(2, 2) + TimetoReformat.Substring(5, 2));
+            }
+            else
+            if (TimetoReformat.Substring(2, 1) == ":")
+            {
+                timeString = (TimetoReformat.Substring(0,2) + TimetoReformat.Substring(3, 2) + TimetoReformat.Substring(6,2));
+            }
+            else
+            {
+                Console.WriteLine("FAIL: Invalid time of " + TimetoReformat);
+                test.Log(Status.Fail, "Invalid time of '" + TimetoReformat + "'");
+            }
+            return timeString;
+        }
+
+        public string ConvertDate(string DatetoConvert)
+        {
```

Comment:
same with this method, break down into smaller methods also remove comments as its hard to read the code

Reply:
Modified ConvertDate method to use a new method (CalculateTheDate) to execute reusable lines of code. Also cleaned up excessive comments and left informative comments.